news/9778.feature.rst

```diff
@@ -0,0 +1,12 @@
+Add an experimental ``always-install-via-wheel`` feature flag. When enabled, it has the
+following effects:
+
+- Installation of source distributions is always done by first building a wheel
+  then installing from it (assuming the ``wheel`` package is installed).
+- ``--no-binary`` now only means "do not download wheels", whereas before it also
+  implied using the legacy ``setup.py install`` and disabling the wheel cache.
+- The use of ``--install-option``, ``--build-option``, ``--global-option`` no longer
+  imply ``--no-binary :all:``.
+- ``--build-option`` and ``--global-option`` are passed to ``setup.py bdist_wheel``
+  during installation, as it was already done when using ``pip wheel``.
+- ``--install-option`` is ignored (because there is no ``setup.py install`` involved)
```

Comment:
Would this render well in the release notes? If not, the details should be moved to documentation.

Reply:
I'll keep note to check how it renders in the release notes. But good point about the documentation!

Reply:
I added a commit with documentation.

Reply:
> Would this render well in the release notes?

https://pip--9778.org.readthedocs.build/en/9778/news/#features -- yes.

Reply:
Although, I'd prefer to say "see [here] for more details" in the changelog TBH.

Reply:
Although, I'd strongly prefer to say something like "see [here] for more details" in the changelog.

The goal of the changelog isn't to communicate *everything* about the change, but only to summarise it in a sentence or two.

Reply:
I agree, assuming there is a "[here]" that reflects the changes. In this case my first attempt to update the doc makes changes spread in different places. But maybe we should add a section dedicated to feature flags?